src/components/form/Form.tsx

```diff
@@ -47,7 +71,17 @@ const Form = () => {
             height={"86px"}
             placeholder={"wpisz jak bardzo lubię wodę..."}
             type={"text"}
-            {...register("watering")}
+            {...register("watering", {
+              maxLength: {
+                value: 512,
+                message: "Opis może zawierać maksymalnie 512 znaków!",
+              },
+            })}
+          />
+          <ErrorMessage
```

Comment:
it's hard to distinguish error message from input label, I'd recommend using some red instead. Not a blocker, though :)